### ISSUE TITLE
multi line FOR-Statement is classified as single line, giving false error

### ISSUE THREAD

**Issue by stereochrome** [Ardem/yii-coding-standard#9]
I have the following input: 

```
for ($i=0;$i<$this->months;$i++) {
  $monthStart = ...
```

and i get the following error:

54 | ERROR | Single line "FOR" must have an expression started from new line. 

i have looked into the sniff and from my understanding it is wrongly thinking of this for-statement as a single line for statement and so it can not understand the code.

i have written the following test code:

```
for ($i=0;$i<10;$i++) 
    doSomething();

for ($i=0;$i<10;$i++) doSomething();

for ($i=0;$i<10;$i++) {
    doSomething();
}
```

the sniff breaks for the second and third for-statement.

i think there is a problem in line 60 of Yii/Sniffs/ControlStructures/SingleLineWithoutBracesSniff.php because the search for the opening curly bracket is done "local" so it stops at the first semikolon. In a for statement the semikolon is used inside the declaration _before_ the opening curly bracket.

simply doing a non-local search does not fix the problem as in the above testcase no error would be found (missing the second for-statement).

maybe i can later write a patch.

**Comment by Ardem** [comment]
Fixed https://github.com/Ardem/yii-coding-standard/commit/9440b6e1e5e6cc7d0e2c2871eb2bcc0f2465cce2
